### ISSUE TITLE
denormalize using `schema.Union` with Immutable data does not denormalize

### ISSUE THREAD

**Issue by n-sviridenko** [paularmstrong/normalizr#255]
# Problem

It doesn't denormalize the entity.

**Input**

Here's how I'm using normalizr:

short:
https://runkit.com/58f68083e5dc270012c70492/58f7a233718530001238d7b9

long:
```js
var { schema, denormalize } = require("normalizr");
var { fromJS } = require("immutable");

const a = new schema.Entity('as');
const b = new schema.Entity('bs');
const ab = new schema.Union({ a, b }, 'type');

const loader = {
    loading: false,
    error: null,
    data: {
        id: 1,
        schema: 'a',
    },
};

const entities = {
    as: {
        '1': {
            id: 1,
            type: 'a',
        },
    },
};

console.log(denormalize(loader, { data: ab }, entities));
console.log(denormalize(fromJS(loader), { data: ab }, fromJS(entities)).toJS());
```

**Output**

```js
{
	"loading": false,
	"error": null,
	"data": {
		"id": 1,
		"type": "a"
	}
}
{
	"loading": false,
	"error": null,
	"data": {
		"id": 1,
		"schema": "a"
	}
}
```

**Comment by n-sviridenko** [comment]
Hello! Is there any news?

**Comment by paularmstrong** [comment]
Fixed by #257
